src/compiler/sourcemap.ts

```diff
@@ -265,6 +305,7 @@ namespace ts {
 
         function toJSON(): RawSourceMap {
             commitPendingMapping();
+            const mappings = (lastMappings ??= decoder.decode(mappingsBuffer.slice(0, mappingsPos)));

```

Comment:
`slice` allocates lol

```suggestion
            const mappings = (lastMappings ??= decoder.decode(mappingsBuffer.subarray(0, mappingsPos)));
```